lib/ayesql/compiler.ex

```diff
@@ -181,7 +189,47 @@ defmodule AyeSQL.Compiler do
   end
 
   @spec create_query(query()) :: Macro.t()
-  defp create_query({name, docs, fragments}) do
+  defp create_query({name, :script, docs, fragments}) do
+    fragments = Macro.escape(fragments)
+
+    quote do
+      @doc AyeSQL.Compiler.gen_docs(unquote(docs), unquote(fragments))
+      @spec unquote(name)(AyeSQL.Core.parameters()) ::
+              {:ok, AyeSQL.Query.t() | term()}
+              | {:error, AyeSQL.Error.t() | term()}
+      @spec unquote(name)(AyeSQL.Core.parameters(), AyeSQL.Core.options()) ::
+              {:ok, AyeSQL.Query.t() | term()}
+              | {:error, AyeSQL.Error.t() | term()}
+      def unquote(name)(params, options \\ [])
+
+      def unquote(name)(params, options) do
+        options = Keyword.merge(__MODULE__.__db_options__(), options)
+
+        {index, options} = Keyword.pop(options, :index, 1)
+        {run?, options} = Keyword.pop(options, :run, true)
+
+        content = AyeSQL.AST.expand(__MODULE__, unquote(fragments))
+        context = AyeSQL.AST.Context.new(index: index)
+
+        with {:ok, %{statement: statement} = query} <-
+               AyeSQL.Core.evaluate(content, params, context) do
+          if run? do
+            statement
+            |> String.split(";", trim: true)
```

Comment:
@dkuku There's parsing outside of the parser for a specific case. This should be handled somehow inside the parser in case it needs to be supported. This raises new questions though:
- Should we split the parser between `normal` and `script` types?
- Show `script` types accept more than one `script` (so more than one generated function) from the same file?

Reply:
If we wan't to implement this feature it needs some discussion around it. And be wrapped in transaction. The blocker I had was that transactions are differently named in postgres comparing to mysql so this would need to be somehow configurable.